src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -294,10 +294,9 @@ ExpectedPrivateKey PrivateKey::Load(const Args &args) {
 	return LoadFrom(args);
 }
 
-ExpectedPrivateKey PrivateKey::Generate(const unsigned int bits, const unsigned int exponent) {
-#ifdef MENDER_CRYPTO_OPENSSL_LEGACY
+ExpectedPrivateKey PrivateKey::Generate() {
 	auto pkey_gen_ctx = unique_ptr<EVP_PKEY_CTX, void (*)(EVP_PKEY_CTX *)>(
-		EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, nullptr), pkey_ctx_free_func);
+		EVP_PKEY_CTX_new_id(EVP_PKEY_ED25519, nullptr), pkey_ctx_free_func);
```

Comment:
Previously, we used `EVP_PKEY_CTX_new_id` for OpenSSL legacy, and `EVP_PKEY_CTX_new_from_name` for new versions. From what I read in the man page we should prefer `EVP_PKEY_CTX_new_from_name` - Is there a reason for this API call to not work for ED25519?

Reply:
> I don't see anything in the man page that says that we should prefer `EVP_PKEY_CTX_new_from_name()`, (...)

There is no explicit mention (you are right), it is rather my interpretation,

Based on 1) that the `id` parameter itself are referred as legacy (from the manual: "Legacy NID or EVP_PKEY type (...) These  are  EVP_PKEY_RSA, EVP_PKEY_RSA_PSS, .(.. ) and  are used by legacy methods.) and 2) that we had already int he code the distinction when generating RSA keys, I made a case for that we should "prefer" it.

Also, this is not really related to the change that we want to introduce (replace the key pair type).

TBH I don't have an strong opinion. I expressed here my view and do as you think is best. I trust you on this.

Reply:
Ah, didn't know about the `id` being considered legacy. I'll go with `EVP_PKEY_CTX_new_id` for legacy and `EVP_PKEY_CTX_new_from_name` for newer then :+1: